pa/src/router/router.js

```diff
@@ -0,0 +1,50 @@
+import VueRouter from "vue-router";
+import AccountsList from "../components/AccountsList";
+import IndicesList from "../components/IndicesList"
+import IndexView from "../components/IndexView"
+import GoalsList from "../components/GoalsList";
+import GoalView from "../components/GoalView";
+import HomePage from "../components/HomePage";
+import Login from "../components/Login";
+import Registration from "../components/Registration";
+import PortfoliosList from "../components/PortfoliosList";
+import PortfolioView from "../components/PortfolioView";
+
+const router = new VueRouter({
+    mode: "history",
+    routes: [
+        {path: "/", name: "home", component: HomePage,},
+        {path: "/login", name: 'login', component: Login, meta: {guest: true}},
+        {path: "/registration", name: "registration", component: Registration, meta: {guest: true}},
+        {path: "/accounts", component: AccountsList, meta: {requiresAuth: true}},
+        {path: "/indices", component: IndicesList, meta: {requiresAuth: true}},
+        {path: "/indices/:id", component: IndexView, meta: {requiresAuth: true}},
+        {path: "/goals", component: GoalsList, meta: {requiresAuth: true}},
+        {path: "/goals/:id", component: GoalView, meta: {requiresAuth: true}},
+        {path: "/portfolios", component: PortfoliosList, meta: { requiresAuth: true }},
+        {path: "/portfolios/:id", component: PortfolioView, meta: { requiresAuth: true }}
+    ]
+});
+
+router.beforeEach((to, from, next) => {
+    if (to.matched.some(record => record.meta.requiresAuth)) {
```

Comment:
it's better to check token's validity on axios requests in base methods not on page change as you can't check if token is valid and not expired only by looking as its presence